src/main/java/com/bobocode/hoverla/bring/context/BeanValidator.java

```diff
@@ -1,18 +1,125 @@
 package com.bobocode.hoverla.bring.context;
 
+import com.bobocode.hoverla.bring.exception.CircularDependencyException;
+import com.bobocode.hoverla.bring.exception.IncorrectBeanNameException;
+import com.bobocode.hoverla.bring.exception.IncorrectBeanTypeException;
+import com.bobocode.hoverla.bring.exception.NoSuchBeanException;
 import com.bobocode.hoverla.bring.exception.NoUniqueBeanException;
+import lombok.extern.slf4j.Slf4j;
 
 import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
 /**
  * Validates beans list to avoid duplicates or other conflicts
  */
+@Slf4j
 public class BeanValidator {
 
     /**
-     * @throws NoUniqueBeanException if there's bean duplicate found
+     * @param beanDefinitions all bean definitions of application context
+     * @throws IllegalArgumentException    if beanDefinitions is not initialized
+     * @throws NoUniqueBeanException       if there's bean duplicate found
+     * @throws IncorrectBeanNameException  if bean name is empty or blank
+     * @throws IncorrectBeanTypeException  if bean type is not suitable for the required type
+     * @throws CircularDependencyException if occurs circular dependency
+     * @throws NoSuchBeanException         if the application context doesn't contain such a bean
      */
     public void validate(List<BeanDefinition> beanDefinitions) {
+        log.debug("Bean validation started");
+        if (beanDefinitions == null) {
+            throw new IllegalArgumentException("Bean definition list is not initialized");
+        }
 
+        beanDefinitions.forEach(beanDefinition -> {
+            beanNameValidation(beanDefinition);
+            dependenciesValidation(beanDefinition, beanDefinitions);
+        });
+        var duplicateNames = beanDefinitions.stream()
```

Comment:
`var` is not aplicable here. It should be used when the return type is obvious (for example with `new` statement)

Reply:
Done